Some cleanup of cross library test imports

justinmc opened the pull request:
There is a lot of cross-library importing in the framework unit tests that I'd like to clean up, see the design doc for more: https://docs.google.com/document/d/1UHxALQqCbmgjnM1RNV9xE2pK3IGyx-UktGX1D7hYCjs/edit?tab=t.0

This PR cleans up a few obvious instances and adds TODOs for others. I created this while doing an investigation for the design doc linked above. I hope that we'll be able to follow up with fixes for all of the problematic tests (tracked in the issue below).

Part of https://github.com/flutter/flutter/issues/177028

justinmc committed:
Unnecessary use of Material localizations

justinmc committed:
This will never build a Material Scrollbar

justinmc committed:
TODOs

justinmc committed:
This will never build a Material Scrollbar, either.

justinmc committed:
Rest of Cupertino TODOs

justinmc committed:
No need for cupertino, probably a result of copying

justinmc committed:
Some Widgets todos

justinmc committed:
SelectableText is a Material widget

justinmc committed:
Unused Cupertino

justinmc committed:
TODO for Cupertino in Widgets

justinmc committed:
Unnecessary Material imports

justinmc committed:
Should have been in Material

justinmc committed:
Colors => Color

justinmc committed:
Unnecessary Material imports

justinmc committed:
Colors => Color

justinmc committed:
Unnecessary Material import(s)

justinmc committed:
I guess I messed up that comment at some point...

justinmc committed:
Unnecessary Material imports

justinmc committed:
Green

justinmc committed:
Unnecessary Material imports

justinmc committed:
Unnecessary Material imports

justinmc committed:
Colors

justinmc committed:
Unnecessary Material import

justinmc committed:
Colors.green

justinmc committed:
Unnecessary Material import

justinmc committed:
Unnecessary Material import

justinmc committed:
Colors

justinmc committed:
Unnecessary Material imports

justinmc committed:
Colors

justinmc committed:
Unnecessary Material import

justinmc committed:
Unnecessary materiallocalizations

justinmc committed:
Unnecessary Material import

justinmc committed:
Colors.orange

justinmc committed:
Unnecessary Material import

justinmc committed:
Unnecessary Material import

justinmc committed:
Colors.red

justinmc committed:
Unnecessary Material imports

justinmc committed:
Unnecessary imports of Material

justinmc committed:
Colors

justinmc committed:
Unnecessary import of Material

justinmc committed:
Colors

justinmc committed:
Unnecessary imports of Material

justinmc committed:
Merge branch 'master' into cross-library-test-imports

justinmc committed:
Analyzer

Renzo-Olivares reviewed (approved):
LGTM w/ some small comments.

justinmc committed:
Formatting

justinmc committed:
Whoops, correct use of Cupertino

justinmc committed:
Merge branch 'master' into cross-library-test-imports

huycozy commented:
I only want to inform that I'm looking at the umbrella issue (and its sub-issues too) and would like to give a hand to complete the remaining work. Hope this PR can be merged more quickly. This will help to determine the remaining tasks for the sub-issues outlined in https://github.com/flutter/flutter/issues/177028. It would be appreciated if completed items can be marked as done there, so that we can check the status of the list better. Thank you! 🙏

justinmc commented:
Thank you for the support @huycozy! I just landed the lint PR https://github.com/flutter/flutter/pull/178693 that was blocking this. I'm going to update this PR and make sure it works with the lint, then everyone should be good to start working on https://github.com/flutter/flutter/issues/177028.

However, there are some test helpers that I would love to get written to make some of these tests easier. For example, @Renzo-Olivares was looking at writing a test helper wrapping EditableText to make it easier to use. The same thing is probably needed for WidgetsApp.

I'll respond on https://github.com/flutter/flutter/issues/177028 and the sub-issues with more details.

justinmc committed:
Merge branch 'master' into cross-library-test-imports

justinmc committed:
Merge branch 'master' into cross-library-test-imports

justinmc committed:
Analyzer

justinmc commented:
Indeed the analyzer has failed due to the lint I added in https://github.com/flutter/flutter/pull/178693. The cross imports that I fixed in this PR now need to be removed from the allowlist. I'll do that now and then it should pass.

<details>

<summary>Analyzer error</summary>

```
╔═╡ERROR #1╞════════════════════════════════════════════════════════════════════
║ Huzzah! The following tests in Widgets no longer contain cross imports!
║   packages/flutter/test/widgets/async_lifecycle_test.dart
║   packages/flutter/test/widgets/multi_view_parent_data_test.dart
║   packages/flutter/test/widgets/view_test.dart
║   packages/flutter/test/widgets/slotted_render_object_widget_test.dart
║   packages/flutter/test/widgets/semantics_2_test.dart
║   packages/flutter/test/widgets/reparent_state_test.dart
║   packages/flutter/test/widgets/transitions_test.dart
║   packages/flutter/test/widgets/parent_data_test.dart
║   packages/flutter/test/widgets/binding_live_test.dart
║   packages/flutter/test/widgets/list_view_fling_test.dart
║   packages/flutter/test/widgets/selectable_text_test.dart
║   packages/flutter/test/widgets/semantics_7_test.dart
║   packages/flutter/test/widgets/opacity_repaint_test.dart
║   packages/flutter/test/widgets/list_view_relayout_test.dart
║   packages/flutter/test/widgets/semantics_4_test.dart
║   packages/flutter/test/widgets/multi_view_binding_test.dart
║   packages/flutter/test/widgets/binding_deferred_first_frame_test.dart
║   packages/flutter/test/widgets/scroll_delegate_test.dart
║   packages/flutter/test/widgets/list_view_correction_test.dart
║   packages/flutter/test/widgets/semantics_zero_surface_size_test.dart
║   packages/flutter/test/widgets/semantics_9_test.dart
║   packages/flutter/test/widgets/icon_data_test.dart
║   packages/flutter/test/widgets/semantics_refactor_regression_test.dart
║   packages/flutter/test/widgets/animated_opacity_repaint_test.dart
║   packages/flutter/test/widgets/semantics_6_test.dart
║   packages/flutter/test/widgets/animated_image_filtered_repaint_test.dart
║   packages/flutter/test/widgets/tree_shape_test.dart
║   packages/flutter/test/widgets/rich_text_test.dart
║   packages/flutter/test/widgets/sensitive_content_error_handling_test.dart
║   packages/flutter/test/widgets/drag_boundary_test.dart
║   packages/flutter/test/widgets/semantics_traversal_test.dart
║   packages/flutter/test/widgets/sensitive_content_unknown_test.dart
║   packages/flutter/test/widgets/overlay_test.dart
║   packages/flutter/test/widgets/semantics_8_test.dart
║   packages/flutter/test/widgets/sliver_fill_viewport_test.dart
║   packages/flutter/test/widgets/wrap_test.dart
║   packages/flutter/test/widgets/overlay_portal_test.dart
║   packages/flutter/test/widgets/semantics_11_test.dart
║   packages/flutter/test/widgets/binding_first_frame_rasterized_test.dart
║   packages/flutter/test/widgets/router_restoration_test.dart
║   packages/flutter/test/widgets/error_widget_test.dart
║   packages/flutter/test/widgets/semantics_checks_test.dart
║   packages/flutter/test/widgets/sensitive_content_test.dart
║   packages/flutter/test/widgets/semantics_10_test.dart
║   packages/flutter/test/widgets/semantics_1_test.dart
║   packages/flutter/test/widgets/sensitive_content_host_test.dart
║   packages/flutter/test/widgets/default_colors_test.dart
║   packages/flutter/test/widgets/multi_view_tree_updates_test.dart
║   packages/flutter/test/widgets/semantics_5_test.dart
║   packages/flutter/test/widgets/clip_test.dart
║   packages/flutter/test/widgets/independent_widget_layout_test.dart
║   packages/flutter/test/widgets/binding_first_frame_developer_test.dart
║   packages/flutter/test/widgets/multi_view_no_implicit_view_binding_test.dart
║   packages/flutter/test/widgets/reparent_state_with_layout_builder_test.dart
║   packages/flutter/test/widgets/semantics_3_test.dart
║   packages/flutter/test/widgets/sliver_appbar_opacity_test.dart
║   packages/flutter/test/widgets/sliver_constrained_cross_axis_test.dart
║   packages/flutter/test/widgets/focus_manager_test.dart
║   packages/flutter/test/widgets/implicit_semantics_test.dart
║   packages/flutter/test/widgets/shrink_wrapping_viewport_test.dart
║ However, they now need to be removed from the
║ knownWidgetsCrossImports list in the script /dev/bots/check_tests_cross_imports.dart.
╚═══════════════════════════════════════════════════════════════════════════════
╔═╡ERROR #2╞════════════════════════════════════════════════════════════════════
║ Huzzah! The following tests in Cupertino no longer contain cross imports!
║   packages/flutter/test/cupertino/action_sheet_test.dart
║ However, they now need to be removed from the
║ knownCupertinoCrossImports list in the script /dev/bots/check_tests_cross_imports.dart.
╚═══════════════════════════════════════════════════════════════════════════════
```

</details>

justinmc committed:
Remove tests from allowlist whose cross import is fixed

justinmc committed:
Now that typing lint has changed, update types on my Colors.
